src/vs/code/browser/workbench/workbench.ts

```diff
@@ -422,8 +430,8 @@ class WorkspaceProvider implements IWorkspaceProvider {
 		return targetHref;
 	}
 
-	private encodeWorkspacePath(uri: URI): string {
-		if (this.config.remoteAuthority && uri.scheme === Schemas.vscodeRemote) {
+	private encodeWorkspacePath(uri: URI, remoteAuthority: string | undefined): string {
+		if (uri.scheme === Schemas.vscodeRemote && isEqualAuthority(this.config.remoteAuthority, remoteAuthority)) {
```

Comment:
This method is not clear to me anymore: what if the `uri` has a different `remoteAuthority` than what is passed in? Don't we have to rewrite the `Uri` then?